src/Hystrix.Dotnet/DateTimeProvider.cs

```diff
@@ -2,11 +2,9 @@
 
 namespace Hystrix.Dotnet
 {
-    public class DateTimeProvider
+    public class DateTimeProvider : IDateTimeProvider
```

Comment:
This is the only real change, everything else is just the ReSharper syntactical suggestions.
This was done in a weird way, we had this class without an interface, and used it for mocking, but didn't had an interface for it, and we weren't actually injecting it, but instantiating on the inside.
I changed it so it has a proper interface, and we are injecting it to all the classes dependent on it.